modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py

```diff
@@ -0,0 +1,487 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module houses classes responsible for storing an axis partition and applying a function to it."""
+
+import pandas
+
+from modin.core.dataframe.pandas.partitioning.axis_partition import (
+    PandasDataframeAxisPartition,
+)
+from .partition import PandasOnRayDataframePartition
+
+import ray
+from ray.util import get_node_ip_address
+
+
+class PandasOnRayDataframeVirtualPartition(PandasDataframeAxisPartition):
+    """
+    The class implements the interface in ``PandasDataframeAxisPartition``.
+
+    Parameters
+    ----------
+    list_of_blocks : list
+        List of ``PandasOnRayDataframePartition`` objects.
+    get_ip : bool, default: False
+        Whether to get node IP addresses to conforming partitions or not.
+    full_axis : bool, default: True
+        Whether or not the virtual partition encompasses the whole axis.
+    call_queue : list, optional
```

Comment:
Does this refer to each physical partition in a list_of_blocks?

Reply:
No, this is referring to the `call_queue` for the virtual partition as a whole.